WordPress/openverse-frontend
# Remove need to pin pnpm version

## Description


Previously in #1385 we pinned the pnpm version to the latest 6.x.x. We did this because version 7 introduced an installation failure when peer dependencies were not being met.

Rather than continue to fight to pin pnpm, it seemed to make sense to actually try to fix the failures it was complaining about. Previously I'd avoided this because I wasn't sure what to do about react being a peer dependency of storybook and use clearly not wanting to declare a dependency on React (it makes no sense). I'm not sure why Storybook needs it considering it works fine without it. Maybe something else in our toolchain is installing it directly :shrug: 

For this PR I went through our pnpm-lock.yaml and identified the packages that were already in there anyway with version that were already being pulled into our node_modules. I added those to our package.json. Finally, I added some ignore rules and some overrides to let pnpm know that we know what we're doing and that the versions we have will be okay (at least, they have been okay so far).

Finally, this PR also pins our pnpm version in `engines` at >=7.0.1 and properly restricts our node version to >=16 <17. Node 17 continues not to play nice with webpack 4 or Nuxt, it seems (I guess the other way around maybe).

All of this has the added benefit of simplifying everywhere that got more complicated by #1385. We can remove the `.pnpm-version` file and all the juggling we had to do to get its values into Dockerfiles and other places it didn't want to be on its own.

## Testing Instructions

Make sure all of CI passes including the docker build etc.

Checkout the branch, delete `node_modules`, `volta install pnpm@7` and run `pnpm install`. Make sure it's successful and doesn't complain about any unmet peer dependencies.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
